Adds config option for allowing webclient.

MrsTonedOne opened the pull request:
Also adds a config option to restrict webclient to byond members only.

The logic for restricting it to byond members only is that its easy to evade with the webclient, locking that behind a 10 dollar pay wall should curtail that mostly.

More advanced systems (like tracking cids by connection type, and cid bans by last 'seeker' connection, and requiring one recent seeker connection) could come later, but they would require special DB edits, so this will work while we flesh it out.

MrsTonedOne committed:
Adds config option for allowing webclient.
Also adds a config option to restrict webclient to byond members only.

Giacom commented on code/modules/client/client procs.dm:
```diff
@@ -122,6 +122,16 @@ var/next_external_rsc = 0
+			del(src)
+			return 0
+		if (config.webclientmembersonly && !is_content_unlocked())
+			src << "Sorry, but the web client is restricted to byond members only."
```
It might make players angry if you don't explain why the web client is behind a paywall. Maybe it could be explained here?

optimumtact commented:
If this goes in is it going to be turned on ?

MrStonedOne commented:
I might wait until I update the server to 509 since rightclick randomly broke (it used to work) and 509 fixes that.

optimumtact commented:
:small_blue_diamond:

MrsTonedOne committed:
Changes webclient member checking to use IsByondMember()

You can disable byond member features in preferences, something the other method checked, but this could lead to somebody not being able to get in to re-enable them.

MrsTonedOne committed:
Fixes compile error in client procs from mismatched ()
